core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -435,6 +436,7 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
   }
 
   private def validateCreateProjectRequest(createProjectRequest: CreateRawlsBillingProjectFullRequest): Future[Unit] = {
+    validateBillingProjectName(createProjectRequest.projectName.value)
```

Comment:
This is not great from a FP and `Future` perspective. The return value is `Future[Unit]`. The expectation is that it returns a Future which could be failure or success. This changes it so it returns a `Future` or throws an exception. This works because the calling code is tolerant. It would be better to return a `Future.failed` as the code below does.